multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ValidateDeployParametersStep.java

```diff
@@ -177,13 +179,18 @@ private void mergeArchive(ProcessContext context, List<FileEntry> archivePartEnt
     private void mergeArchive(ProcessContext context, List<FileEntry> archivePartEntries, BigInteger archiveSize) {
         ArchiveStreamWithName archiveStreamWithName = getMergedArchiveStreamCreator(archivePartEntries, archiveSize).createArchiveStream();
         try {
+            LazyArchiveInputStream lazyArchiveInputStream = (LazyArchiveInputStream) archiveStreamWithName.getArchiveStream();
+            FileMimeTypeValidator.validateInputStreamMimeType(lazyArchiveInputStream.getCurrentInputStream(),
+                                                              archiveStreamWithName.getArchiveName());
```

Comment:
I do not think this validation should be here, this way we are only checking the first part of the archive (before merging).

Use the stream after the merge is done.